demos/PDE_constrained_optimisation/PDE_constrained_field.py

```diff
@@ -54,7 +52,7 @@
 x0, y0 = 0.75, 0.5
 w = 0.1
 r2 = (x - x0) ** 2 + (y - y0) ** 2
-T0.interpolate(exp(-r2 / w**2))
+T0 = interpolate(exp(-r2 / w**2), Q)
```

Comment:
I don't understand this change. `T0` is being initialised on line 37, then discarded and replaced by the result of the `interpolate` function on Line 55, whereas before `interpolate` was being applied to the existing `T0` object. Its fairly inconsequential here, you only lose the "Initial Temperature" name, but it may be more consequential in other situations.

Reply:
This was me being silly. Good spot. All fixed up.